frontend/src/metabase/admin/permissions/utils/graph/partial-updates.ts

```diff
@@ -0,0 +1,97 @@
+import _ from "underscore";
+
+import type { GroupsPermissions } from "metabase-types/api";
+
+type AdvancedPermissions = Record<
+  string,
+  { group_id: number; [key: string]: unknown }[] | undefined
+>;
+
+interface AdvancedPermissionsData {
+  modifiedGroupIds: string[];
+  permissions: AdvancedPermissions;
+}
+
+// utils for dealing with partial graph updates
+
+function getModifiedGroupIdsInGraph(
+  groupIds: string[],
+  originalDataPermissions: GroupsPermissions,
+  newDataPermissions: GroupsPermissions,
+) {
+  // find only the groupIds that have had some kind of modification made in their permissions graph
+  return groupIds.filter(groupId => {
+    return !_.isEqual(
+      newDataPermissions[groupId as unknown as number],
+      originalDataPermissions[groupId as unknown as number],
+    );
+  });
+}
+
+function getModifiedAdvancedPermissions(
+  advancedPermissions: AdvancedPermissions,
+  modifiedGroupIds: string[],
+) {
+  const groupIds = new Set(modifiedGroupIds);
+
+  return Object.fromEntries(
+    Object.entries(advancedPermissions).map(([key, items]) => {
+      const modified = (items || []).filter(item =>
+        groupIds.has(`${item.group_id}`),
```

Comment:
maybe make this cast a bit more explicit?
```suggestion
        groupIds.has(String(item.group_id})),
```

Reply:
ended up removing this function